### PR TITLE
feat(policy): 1277 add Resource Mapping Group to objects proto

### LINKED ISSUE #1277: Implement protos and service methods for Resource Mapping Groups operations
## Acceptance Criteria
- Create object protos
- Create new service proto
- Implement service proto handlers and update existing `sqlc` logic accordingly
- Add integration tests for new functionality

### PR DESCRIPTION
**Pull request by ryanulit**
Resolves #1277

### CODE REVIEW

**Comment by ryanulit** [comment]
Note: this PR does not actually close #1277 and is 1/2 PRs, but the check requires a GH keyword to be used so 🤷

File: service/policy/objects.proto [hunk 1]
```diff
@@ -221,6 +221,14 @@ message SubjectProperty {
   string external_value = 2 [(buf.validate.field).required = true];
 }
 
+message ResourceMappingGroup {
```
Comment (jakedoublev) [inline service/policy/objects.proto:224]:
```suggestion
ResourceMappingGroups are namespaced collections of resource mappings associated under a common group name.
message ResourceMappingGroup {
```

File: service/policy/objects.proto [hunk 2]
```diff
@@ -221,6 +221,14 @@ message SubjectProperty {
...
+  string id = 1;
+
+  string namespace_id = 2 [(buf.validate.field).required = true];
```
Comment (jakedoublev) [inline service/policy/objects.proto:227]:
```suggestion
  The namespace holding the group and its resource mappings within
  string namespace_id = 2 [(buf.validate.field).required = true];
```

File: service/policy/objects.proto [hunk 3]
```diff
@@ -221,6 +221,14 @@ message SubjectProperty {
...
+
+  string name = 3 [(buf.validate.field).required = true];
```
Comment (jakedoublev) [inline service/policy/objects.proto:229]:
```suggestion
  The common name for the group collection of resource mappings. RM Groups within a single namespace must each have a unique name.
  string name = 3 [(buf.validate.field).required = true];
```

**Review by jakedoublev — APPROVED** [review]
LGTM

**Pull request merged**